### ISSUE TITLE
Composer status failing and claims filesize is failing on node_modules folder

### ISSUE THREAD

**Issue by Nek-** [composer/composer#8898]
Output of `composer diagnose`:

```
Checking composer.json: WARNING
require.bitbag/elasticsearch-plugin : unbound version constraints (dev-master) should be avoided
Checking platform settings: OK
Checking git settings: OK
Checking http connectivity to packagist: OK
Checking https connectivity to packagist: OK
Checking github.com rate limit: OK
Checking disk free space: OK
Checking pubkeys: 
Tags Public Key Fingerprint: 57815BA2 7E54DC31 7ECC7CC5 573090D0  87719BA6 8F3BB723 4E5D42D0 84A14642
Dev Public Key Fingerprint: 4AC45767 E5EC2265 2F0C1167 CBBB8A2B  0C708369 153E328C AD90147D AFE50952
OK
Checking composer version: OK
Composer version: 1.10.6
PHP version: 7.4 - Package overridden via config.platform (actual: 7.4.5)
PHP binary path: /usr/bin/php7.4
OpenSSL version: OpenSSL 1.1.1c  28 May 2019
```

When I run this command:

```
composer status
```

I get the following output:

```
  [ErrorException]                            
  filesize(): stat failed for ./node_modules
```

And I expected this to happen: no failure

Record of my terminal : https://asciinema.org/a/1v9Vhl00OXxzERehNknErmZMn

It may be related to my environment but since I can't figure out how or why I open an issue. This is not critical it's "just" on status.

Please notice it does not seems to be related to the very last version of composer because I did an upgrade from composer `1.8.5̀` after seeing this issue which still occurs on the last version for me.

**Comment by jacquesbh** [comment]
Do you have the same issue if you remove the `node_modules` directory?

**Comment by stof** [comment]
Can you run the command in verbose mode (with `-v`) so that the statck trace of the exception is displayed ?

**Comment by Nek-** [comment]
```
composer status -v

                                              
  [ErrorException]                            
  filesize(): stat failed for ./node_modules  
                                              

Exception trace:
 () at phar:///usr/local/bin/composer/src/Composer/Package/Comparer/Comparer.php:113
 Composer\Util\ErrorHandler::handle() at n/a:n/a
 filesize() at phar:///usr/local/bin/composer/src/Composer/Package/Comparer/Comparer.php:113
 Composer\Package\Comparer\Comparer->doTree() at phar:///usr/local/bin/composer/src/Composer/Package/Comparer/Comparer.php:69
 Composer\Package\Comparer\Comparer->doCompare() at phar:///usr/local/bin/composer/src/Composer/Downloader/FileDownloader.php:307
 Composer\Downloader\FileDownloader->getLocalChanges() at phar:///usr/local/bin/composer/src/Composer/Command/StatusCommand.php:100
 Composer\Command\StatusCommand->execute() at phar:///usr/local/bin/composer/vendor/symfony/console/Command/Command.php:241
 Symfony\Component\Console\Command\Command->run() at phar:///usr/local/bin/composer/vendor/symfony/console/Application.php:843
 Symfony\Component\Console\Application->doRunCommand() at phar:///usr/local/bin/composer/vendor/symfony/console/Application.php:193
 Symfony\Component\Console\Application->doRun() at phar:///usr/local/bin/composer/src/Composer/Console/Application.php:254
 Composer\Console\Application->doRun() at phar:///usr/local/bin/composer/vendor/symfony/console/Application.php:117
 Symfony\Component\Console\Application->run() at phar:///usr/local/bin/composer/src/Composer/Console/Application.php:103
 Composer\Console\Application->run() at phar:///usr/local/bin/composer/bin/composer:56
 require() at /usr/local/bin/composer:24

status [-v|vv|vvv|--verbose]
```

If I remove the node_modules I have the *exact* same issue (highlighing the node_modules folder that does not exist anymore!)

**Comment by stof** [comment]
This looks weird. `filesize` is called only when `is_dir` returned false.

**Comment by stof** [comment]
@Nek- is this happening in a shared folder of a VM ?

**Comment by Nek-** [comment]
It's happening on a standard computer.

**Comment by Seldaek** [comment]
Do you have Composer plugins installed? Why is it even looking at the node_modules directory?

**Comment by Nek-** [comment]
No plugins. I have no idea why it is looking for the node_modules directory. I tried to remove it but the issue is still here... So I assume it's related to another node_modules repository than the one I have on my pwd.

**Comment by jacquesbh** [comment]
This issue seems to be fixed between 1.10.9 and 1.10.13.

```
$ sf composer status -v

  [ErrorException]
  filesize(): stat failed for ./node_modules

$ sf composer status -v --no-plugins

  [ErrorException]
  filesize(): stat failed for ./node_modules

$ sf php -v
PHP 7.4.7 (cli) (built: Jun 12 2020 00:00:24) ( NTS )

$ sf composer --version
Composer version 1.10.9 2020-07-16 12:57:00

$ sf composer self-update
Updating to version 1.10.13 (stable channel).
   Downloading (100%)
Use composer self-update --rollback to return to version 1.10.9

$ sf composer status -v
You have changes in the following dependencies:
…
```

(`sf` for `symfony` binary)

**Comment by Seldaek** [comment]
Fixed by https://github.com/composer/composer/commit/e6b45e853aad1036118273c69c69e88138480f42 then I assume
